### ISSUE TITLE
Not able to dowload all stored data to text for N.2 with N > 10000

### ISSUE THREAD

**Issue by edgarcosta** [LMFDB/lmfdb#6132]
For example https://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/19720/2/

Edit: previous link https://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/19720/2/a/

**Comment by AndrewVSutherland** [comment]
If I click on the "Download all stored data" on the page you linked to I get a file with contents
```
# Stored data for newspace 19720.2.a, downloaded from the LMFDB on 12 August 2024.
{"ALdims": [26, 29, 29, 27, 30, 25, 26, 32, 29, 27, 28, 29, 28, 26, 28, 29], "Nk2": 78880, "analytic_conductor": 157.4649927859649, "char_conductor": 1, "char_degree": 1, "char_is_real": true, "char_orbit_index": 1, "char_orbit_label": "a", "char_order": 1, "char_parity": 1, "char_values": [19720, 1, [14791, 9861, 15777, 10441, 6121], [1, 1, 1, 1, 1]], "conrey_index": 1, "cusp_dim": 3225, "dim": 448, "eis_dim": 31, "eis_new_dim": 0, "hecke_orbit_code": 33574152, "label": "19720.2.a", "level": 19720, "level_is_powerful": false, "level_is_prime": false, "level_is_prime_power": false, "level_is_prime_square": false, "level_is_square": false, "level_is_squarefree": false, "level_primes": [2, 5, 17, 29], "level_radical": 4930, "mf_dim": 3256, "mf_new_dim": 448, "plus_dim": 216, "prim_orbit_index": 1, "relative_dim": 448, "sturm_bound": 6480, "trace_display": [0, -8, 0, -16], "traces": [448, 0, -8, 0, 0, 0, -16, 0, 448, 0, 8, 0, -16, 0, 0, 0, 8, 0, -8, 0, 0, 0, 0, 0, 448, 0, -32, 0, 0, 0, -64, 0, 16, 0, 8, 0, -8, 0, -16, 0, -8, 0, -64, 0, 0, 0, 0, 0, 408, 0, 0, 0, 32, 0, 0, 0, -16, 0, 16, 0, -16, 0, -64, 0, -24, 0, -8, 0, -48, 0, 0, 0, 16, 0, -8, 0, -16, 0, -16, 0, 464, 0, 8, 0, 4, 0, 0, 0, 16, 0, -32, 0, -64, 0, 0, 0, 0, 0, 40, 0, 56, 0, 0, 0, 16, 0, 8, 0, 0, 0, 0, 0, 64, 0, 40, 0, -32, 0, 0, 0, 464, 0, 0, 0, 0, 0, -48, 0, 48, 0, 24, 0, -32, 0, 0, 0, 80, 0, 48, 0, 32, 0, 32, 0, 0, 0, -72, 0, 8, 0, -64, 0, 24, 0, 16, 0, -64, 0, 0, 0, 0, 0, -136, 0, 0, 0, -16, 0, 520, 0, -24, 0, -16, 0, -16, 0, 80, 0, -16, 0, 32, 0, 0, 0, -16, 0, 0, 0, -64, 0, 128, 0, 32, 0, 0, 0, 48, 0, -64, 0, -16, 0, 0, 0, 32, 0, 96, 0, -16, 0, -56, 0, 112, 0, 0, 0, 48, 0, 64, 0, 0, 0, -16, 0, 448, 0, 40, 0, -72, 0, 96, 0, 24, 0, -16, 0, -64, 0, -48, 0, 0, 0, -24, 0, 0, 0, -320, 0, 32, 0, -24, 0, 48, 0, 0, 0, 72, 0, -112, 0, 0, 0, 80, 0, -24, 0, 144, 0, -16, 0, -48, 0, -64, 0, 8, 0, 0, 0, -96, 0, -88, 0, 72, 0, 0, 0, 112, 0, 448, 0, -48, 0, -64, 0, 0, 0, 0, 0, 64, 0, -96, 0, 208, 0, 0, 0, -128, 0, 48, 0, 32, 0, -40, 0, 40, 0, -104, 0, 0, 0, 64, 0, 0, 0, -16, 0, 112, 0, 16, 0, 88, 0, 56, 0, -48, 0, 128, 0, 176, 0, -32, 0, -320, 0, 16, 0, 8, 0, 120, 0, 32, 0, 104, 0, 48, 0, -48, 0, 64, 0, 392, 0, 104, 0, -24, 0, -176, 0, 24, 0, 160, 0, -64, 0, 0, 0, 32, 0, -120, 0, 160, 0, 0, 0, -128, 0, 16, 0, -24, 0, 16, 0, 16, 0, 0, 0, -48, 0, 48, 0, 48, 0, -112, 0, 0, 0, 160, 0, -96, 0, 176, 0, -128, 0, 0, 0, 32, 0, -48, 0, -72, 0, 160, 0, 8, 0, -144, 0, -96, 0, -16, 0, 0, 0, -24, 0, 80, 0, -352, 0, 456, 0, 288, 0, -48, 0, 272, 0, 72, 0, 96, 0, 0, 0, 0, 0, -8, 0, 0, 0, -184, 0, -128, 0, 0, 0, -32, 0, -176, 0, 0, 0, 192, 0, -8, 0, 0, 0, 160, 0, -128, 0, 272, 0, 40, 0, -192, 0, 80, 0, 184, 0, 12, 0, 0, 0, 8, 0, -288, 0, -80, 0, 48, 0, 48, 0, -8, 0, -128, 0, -176, 0, 80, 0, 56, 0, 32, 0, 0, 0, 16, 0, -168, 0, 0, 0, 0, 0, 368, 0, 496, 0, -144, 0, -16, 0, 304, 0, 248, 0, -48, 0, -32, 0, -16, 0, -296, 0, 32, 0, 0, 0, 160, 0, 144, 0, -136, 0, -240, 0, -48, 0, 112, 0, -24, 0, 64, 0, -16, 0, -32, 0, 48, 0, 0, 0, -128, 0, 128, 0, 32, 0, -96, 0, -120, 0, 216, 0, -128, 0, 112, 0, -32, 0, -24, 0, 96, 0, 0, 0, -56, 0, 136, 0, 32, 0, -144, 0, 0, 0, 128, 0, -432, 0, 80, 0, -64, 0, -72, 0, -224, 0, 320, 0, 448, 0, -192, 0, -24, 0, -96, 0, 304, 0, -160, 0, -168, 0, -48, 0, 24, 0, -40, 0, -96, 0, 16, 0, -80, 0, -48, 0, -88, 0, 0, 0, 208, 0, 8, 0, 56, 0, 0, 0, -48, 0, 0, 0, 32, 0, -80, 0, 104, 0, -32, 0, -136, 0, -128, 0, 480, 0, 8, 0, 24, 0, 16, 0, 56, 0, -32, 0, 384, 0, 0, 0, -48, 0, -64, 0, 8, 0, -272, 0, 0, 0, 80, 0, -144, 0, -176, 0, 160, 0, 96, 0, 256, 0, 176, 0, 112, 0, -192, 0, 0, 0, -288, 0, 304, 0, 0, 0, -144, 0, -208, 0, -240, 0, -344, 0, 176, 0, -224, 0, -16, 0, 120, 0, 24, 0, 160, 0, 144, 0, 128, 0, -144, 0, 192, 0, 200, 0, -320, 0, 20, 0, 160, 0, 16, 0, 0, 0, 0, 0, -64, 0, 80, 0, -112, 0, -160, 0, 0, 0, 88, 0, 152, 0, 128, 0, 64, 0, -48, 0, 0, 0, 32, 0, 32, 0, 352, 0, 144, 0, 0, 0, -272, 0, -264, 0, 320, 0, -208, 0, 80, 0, 152, 0, 16, 0, 48, 0, 240, 0, 16, 0, -72, 0, 56, 0, -528, 0, 56, 0, 88, 0, 224, 0, 352, 0, 448, 0, -144, 0, 32, 0, -224, 0, -64, 0, -176, 0, 216, 0, 80, 0, 152, 0, -56, 0, 192, 0, -96, 0, 16, 0, -8, 0, 240, 0, 240, 0, 112, 0, 8, 0, -144, 0, 64, 0, -120, 0, 88, 0, 64, 0, 304, 0, 96, 0, -184, 0, -216, 0, 80, 0, -400, 0, 8, 0, -144, 0, 176, 0, 32, 0, -504, 0, 360, 0, 112, 0, 400, 0, 0, 0, 32, 0, -96, 0, -64, 0, 112, 0, -8, 0, -16, 0, 64, 0, 232, 0, 512, 0, 0, 0, 392, 0, 16, 0, -32, 0, -32, 0, 64, 0, 104, 0, -96, 0, 80, 0, -456, 0, 80, 0, 32, 0, 48, 0, 624, 0, -360, 0, -72, 0, 16, 0, 0, 0, -344, 0, -80, 0, -16, 0, 352, 0, 144, 0, -48, 0, -144, 0, -112, 0, 144, 0, 16, 0, 48, 0, 48, 0, -48, 0, -232, 0, 448, 0], "weight": 2, "weight_parity": 1, "newforms": ["19720.2.a.a", "19720.2.a.b", "19720.2.a.c", "19720.2.a.d", "19720.2.a.e", "19720.2.a.f", "19720.2.a.g", "19720.2.a.h", "19720.2.a.i", "19720.2.a.j", "19720.2.a.k", "19720.2.a.l", "19720.2.a.m", "19720.2.a.n", "19720.2.a.o", "19720.2.a.p", "19720.2.a.q", "19720.2.a.r", "19720.2.a.s"], "oldspaces": [[17, 1, 1, 16], [20, 1, 1, 8], [29, 1, 1, 16], [34, 1, 1, 12], [40, 1, 1, 4], [58, 1, 1, 12], [68, 1, 1, 8], [85, 1, 1, 8], [116, 1, 1, 8], [136, 1, 1, 4], [145, 1, 1, 8], [170, 1, 1, 6], [232, 1, 1, 4], [290, 1, 1, 6], [340, 1, 1, 4], [493, 1, 1, 8], [580, 1, 1, 4], [680, 1, 1, 2], [986, 1, 1, 6], [1160, 1, 1, 2], [1972, 1, 1, 4], [2465, 1, 1, 4], [3944, 1, 1, 2], [4930, 1, 1, 3], [9860, 1, 1, 2]]}
```
What are you expecting?

**Comment by edgarcosta** [comment]
Sorry, I meant to point to https://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/19720/2/

**Comment by AndrewVSutherland** [comment]
Ah, it looks like `download_full_space` needs to be updated to handle cases where not all fo the data is available:
```
Exception on /ModularForm/GL2/Q/holomorphic/download_full_space/19720.2 [GET]
Traceback (most recent call last):
  File "/home/sage/sage-9.7/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sage/sage-9.7/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/flask/app.py", line 1519, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sage/sage-9.7/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/flask/app.py", line 1517, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sage/sage-9.7/local/var/lib/sage/venv-python3.10/lib/python3.10/site-packages/flask/app.py", line 1503, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/lmfdb/lmfdb-git-web/lmfdb/classical_modular_forms/main.py", line 657, in download_full_space
    return CMF_download().download_full_space(label)
  File "/home/lmfdb/lmfdb-git-web/lmfdb/classical_modular_forms/download.py", line 359, in download_full_space
    data['newspaces'] = [spc['label'] for spc, forms in space.decomp]
  File "/home/lmfdb/lmfdb-git-web/lmfdb/classical_modular_forms/download.py", line 359, in <listcomp>
    data['newspaces'] = [spc['label'] for spc, forms in space.decomp]
KeyError: 'label'
```
I will take a look (this should be an easy fix).

**Comment by AndrewVSutherland** [comment]
This will be fixed by #6133 (the bug was actually in WebGamma1Space not the download function).
